vector/src/main/java/im/vector/app/features/home/NewHomeDetailFragment.kt

```diff
@@ -379,6 +386,24 @@ class NewHomeDetailFragment :
         }
     }
 
+    private fun refreshUnreadCounterBadge(
+            spacesNotificationCount: RoomAggregateNotificationCount,
+            hasPendingSpaceInvites: Boolean,
+    ) {
+        val badgeState = if (hasPendingSpaceInvites && spacesNotificationCount.notificationCount == 0) {
+            UnreadCounterBadgeView.State.Text(
+                    text = "!",
+                    highlighted = true,
+            )
+        } else {
+            UnreadCounterBadgeView.State.Count(
+                    count = spacesNotificationCount.notificationCount,
+                    highlighted = spacesNotificationCount.isHighlight || hasPendingSpaceInvites,
+            )
+        }
+        views.spacesUnreadCounterBadge.render(badgeState)
```

Comment:
Perhaps the viewmodel could return the correct badge state rather than calculating it in the view?

Reply:
Hum good idea! let me try to update.

Reply:
@jonnyandrew I have extracted the logic inside a UseCase and added unit tests on it. See https://github.com/vector-im/element-android/pull/8159/commits/2bd0126523cca2b15ce7d89a9e18fd5888fdf6bc